apps/api/src/audio/audio.controller.ts

```diff
@@ -0,0 +1,142 @@
+import {
+  BadRequestException,
+  Body,
+  Controller,
+  Post,
+  UploadedFile,
+  UseGuards,
+  UseInterceptors,
+} from "@nestjs/common";
+import { FileInterceptor } from "@nestjs/platform-express";
+import {
+  ApiBody,
+  ApiConsumes,
+  ApiOperation,
+  ApiResponse,
+  ApiSecurity,
+  ApiTags,
+} from "@nestjs/swagger";
+import { ApiKeyGuard } from "src/projects/guards/apikey.guard";
+import { AudioService } from "./audio.service";
+
+@ApiTags("audio")
+@ApiSecurity("apiKey")
+@UseGuards(ApiKeyGuard)
+@Controller("audio")
+export class AudioController {
+  constructor(private readonly audioService: AudioService) {}
+
+  @Post("transcribe")
+  @UseInterceptors(FileInterceptor("file"))
+  @ApiConsumes("multipart/form-data")
+  @ApiBody({
+    description: "Audio file to transcribe",
+    type: "multipart/form-data",
+    schema: {
+      type: "object",
+      properties: {
+        file: {
+          type: "string",
+          format: "binary",
+          description: "Audio file (MP3, WAV, MP4, MPEG, MPGA, M4A, or WEBM)",
+        },
+        format: {
+          type: "string",
+          enum: ["mp3", "wav", "mp4", "mpeg", "mpga", "m4a", "webm"],
+          description:
+            "Audio format (optional, will be auto-detected if not provided)",
+        },
+      },
+      required: ["file"],
+    },
+  })
+  @ApiOperation({
+    summary: "Transcribe audio to text",
+    description:
+      "Upload an audio file and get its transcription. Supports MP3, WAV, MP4, MPEG, MPGA, M4A, and WEBM formats.",
+  })
+  @ApiResponse({
+    status: 400,
+    description: "Invalid audio file or format",
+  })
+  async transcribeAudio(
+    @UploadedFile() file: any,
+    @Body("format") format?: string,
+  ) {
+    this.validateAudioFile(file);
+    const audioFormat = this.determineAudioFormat(file, format);
+    return await this.audioService.transcribeAudio(file.buffer, audioFormat);
+  }
+
+  private validateAudioFile(file: any): void {
+    if (!file) {
+      throw new BadRequestException("No audio file provided");
+    }
+
+    const allowedMimeTypes = [
```

Comment:
we have an existing `mime-types` library that does most of this for you:
- maps filename -> mime type
- maps mime types -> file extensions (to resolve audio/mp4 + video/mp4, etc)